lib/kafka_consumer/offset_server.ex

```diff
@@ -0,0 +1,145 @@
+defmodule KafkaConsumer.OffsetServer do
+  @type offset :: number
+  @type parition :: number
+  @type noreply :: {:noreply, offset}
+  @type reply :: {:reply, offset, offset}
+  @type topic :: String.t
+
+  @callback increment(topic) :: noreply
+  @callback decrement(topic) :: noreply
+  @callback get(topic) :: reply
+  @callback save(topic) :: reply
+  @callback start_link(topic, parition) :: {:ok, pid}
+
+
+  defmacro __using__(_) do
+    quote do
+      @type offset :: number
+      @type partition :: number
+      @type noreply :: {:noreply, offset}
+      @type reply :: {:reply, offset, offset}
+      @type topic :: String.t
+
+      @moduledoc """
+      Holds the last processed offset
+      All servers are registerd with an atom of the topic
+
+      `start_link("that.one.topic")` will register a server under `:that_one_topic`
+      """
+      use GenServer
+
+      @behaviour KafkaConsumer.OffsetServer
+
+      @doc """
+      Starts a OffsetServer for the given topic
+      """
+      @spec start_link(topic, partition) :: {:ok, pid}
+      def start_link(topic, partition) do
+        name = make_server_name(topic)
+        GenServer.start_link(
+          __MODULE__,
+          get_latest_offset(topic, partition),
+          name: name,
+          id: name
+        )
+      end
+
+      @doc """
+      increments the offset
+      """
+      @spec increment(topic) :: noreply
+      def increment(topic) do
+        GenServer.cast(make_server_name(topic), :increment)
+      end
+
```

Comment:
Add partition number for increment/decrement/get and save functions too, please.